py3status/modules/arch_updates.py

```diff
@@ -50,83 +39,87 @@ class Py3status:
     cache_timeout = 600
     format = None
     hide_if_zero = False
-    include_aur = False
+
+    class Meta:
+        deprecated = {"remove": [{"param": "include_aur", "msg": "obsolete"}]}
 
     def post_config_hook(self):
-        if not self.format:
-            if not self.include_aur:
-                self.format = FORMAT_PACMAN_ONLY
-            else:
-                self.format = FORMAT_PACMAN_AND_AUR
-        self.include_aur = self.py3.format_contains(self.format, "aur")
-        self.include_pacman = self.py3.format_contains(self.format, "pacman")
-        if self.py3.format_contains(self.format, "total"):
-            self.include_aur = True
-            self.include_pacman = True
-
-        if self.include_aur:
-            aurs = ["auracle", "yay", "cower"]
-            command = self.py3.check_commands(aurs)
-            if command:
-                self._check_aur_updates = getattr(
-                    self, "_check_aur_updates_{}".format(command)
-                )
+        helper = {
+            "pacman": self.py3.check_commands(["checkupdates"]),
+            "aur": self.py3.check_commands(["auracle", "yay", "cower"]),
+        }
+        if self.format:
+            placeholders = self.py3.get_placeholders_list(self.format)
+            if "total" in placeholders:
+                pass
+            elif not any(helper.values()):
+                raise Exception(STRING_NOT_INSTALLED.format("pacman, aur"))
             else:
-                self.include_aur = False
-                self.py3.notify_user(STRING_NOT_INSTALLED.format(", ".join(aurs)))
-
-    def arch_updates(self):
-        pacman_updates = aur_updates = total = None
-
-        if self.include_pacman:
-            pacman_updates = self._check_pacman_updates()
-
-        if self.include_aur:
-            aur_updates = self._check_aur_updates()
-
-        if pacman_updates is not None or aur_updates is not None:
-            total = (pacman_updates or 0) + (aur_updates or 0)
-
-        if self.hide_if_zero and total == 0:
-            full_text = ""
```

Comment:
I wanted to mention a trivial bug here. `hide_if_zero` does not work correctly when we're offline because `total` would be `None` instead of `0`. New code accounts for both. It's not something users will see anyway.